Loop/Managers/Alerts/LoopSettingsAlerter.swift

```diff
@@ -10,7 +10,8 @@ import LoopKit
 import LoopCore
 
 protocol LoopSettingsAlerterDelegate: AnyObject {
-    var settings: LoopSettings { get set }
+    var settings: LoopSettings { get }
+    @discardableResult func mutateSettings(_ changes: (_ settings: inout LoopSettings) -> Void) -> LoopSettings
```

Comment:
I thought of providing a computed property for `indefiniteWorkoutOverrideEnabledDate` in LoopDataManager, but I believe this still will have thread issues.